ksql-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -279,11 +292,36 @@ private static boolean isTopicDeleteEnabled(final AdminClient adminClient) {
           .orElse(true);
 
     } catch (final Exception e) {
-      log.error("Failed to initialize TopicClient: {}", e.getMessage());
+      LOG.error("Failed to initialize TopicClient: {}", e.getMessage());
       throw new KsqlException("Could not fetch broker information. KSQL cannot initialize", e);
     }
   }
 
+  private static Config getConfig(final AdminClient adminClient) {
```

Comment:
if you make this and `isTopicDeleteEnabled` no static you can access the adminClient as a field.

Though some would say calling a non-static function from the constructor is dangerous as the object is not fully constructed... I would say that's what unit tests are for! :D